ci-operator/step-registry/single-node/conf/aws/single-node-conf-aws-commands.sh

```diff
@@ -14,6 +14,18 @@ SINGLE_NODE_AWS_INSTANCE_TYPE="${SINGLE_NODE_AWS_INSTANCE_TYPE:-${DEFAULT_AWS_IN
 
 echo "Updating install-config.yaml to a single ${SINGLE_NODE_AWS_INSTANCE_TYPE} control plane node and 0 workers"
 
+# RHEL9 based images do not contain pip3, we need to install it. Multiple jobs rely on the installer image
+# so simply using something like upi-installer will break things since some jobs use stable payload which
+# does not include upi-installer.
+OS_VER=$(awk -F= '/^VERSION_ID=/ { print $2 }' /etc/os-release | tr -d '"' | cut -f1 -d'.')
+if [[ ${OS_VER} == "9" ]]; then
+    echo "Detected RHEL9, installing pip"
+    curl -sLO https://bootstrap.pypa.io/get-pip.py
+    python get-pip.py
+    export PATH=$PATH:/output/.local/bin
+fi
```

Comment:
This step already uses upi-installer? https://github.com/openshift/release/blob/master/ci-operator/step-registry/single-node/conf/aws/single-node-conf-aws-ref.yaml#L3

Which includes pyyaml https://github.com/openshift/installer/blob/master/images/installer/Dockerfile.upi.ci

Reply:
Yeah :( I had misunderstood the reliance of other jobs on the installer image using our SNO setup steps, for now I think we should use this fix and we'll revisit these steps because I think the `upi-installer` is ideal for these sort of steps.